tools/lkl/lib/virtio_net_tap.c

```diff
@@ -0,0 +1,127 @@
+/*
+ * tun/tap based virtual network interface feature for LKL
+ * Copyright (c) 2015,2016 Ryo Nakamura, Hajime Tazaki
+ *
+ * Author: Ryo Nakamura <upa@wide.ad.jp>
+ *         Hajime Tazaki <thehajime@gmail.com>
+ *         Octavian Purdila <octavian.purdila@intel.com>
+ */
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <errno.h>
+#include <string.h>
+#include <unistd.h>
+#include <sys/types.h>
+#include <sys/uio.h>
+#include <sys/socket.h>
+#include <sys/stat.h>
+#include <fcntl.h>
+#include <poll.h>
+#include <net/if.h>
+#include <linux/if_tun.h>
+#include <sys/ioctl.h>
+
+#include <lkl_host.h>
+
+struct lkl_netdev_tap {
+	struct lkl_netdev dev;
+	int fd;
+};
+
+
+static int net_tx(struct lkl_netdev *nd, void *data, int len)
+{
+	int ret;
+	struct lkl_netdev_tap *nd_tap = (struct lkl_netdev_tap *) nd;
+
+	ret = write(nd_tap->fd, data, len);
+	if (ret <= 0 && errno == -EAGAIN)
```

Comment:
This might be a bug in the original code, but I guess it should be `EAGAIN` instead of `-EAGAIN`.